src/carma-platform/run_all_control_analysis.py

```diff
@@ -0,0 +1,82 @@
+from pathlib import Path
+from typing import Dict
+from guidance_scripts import (
+    run_crosstrack_analysis,
+    run_turn_accuracy_analysis,
+    run_acceleration_comfort_analysis,
+)
+from run_all_analysis import run_all_analysis
+import argparse
+import argcomplete
+
+
+def analyze_mcap_file_for_control_analysis(
+    mcap_path: Path, output_dir: Path, data_dir: Path, plots_dir: Path
+) -> Dict:
+    """Extract single MCAP file and run all control analysis on it"""
+    try:
+        analysis_stats = {}
+        # 1. Cross_track analysis
+        try:
+            stats, _, _, _ = run_crosstrack_analysis(mcap_path, data_dir, plots_dir)
+            analysis_stats["run_crosstrack_analysis"] = stats
+        except:
+            analysis_stats["run_crosstrack_analysis"] = None
+
+        # 2. Turn accuracy analysis by spline fitting
+        try:
+            stats, _, _, _ = run_turn_accuracy_analysis(mcap_path, data_dir, plots_dir)
+            analysis_stats["run_turn_accuracy_analysis"] = stats
+        except:
+            analysis_stats["run_turn_accuracy_analysis"] = None
+
+        # 3. Turn accuracy analysis by spline fitting
+        try:
+            stats, _, _, _ = run_acceleration_comfort_analysis(
+                mcap_path, data_dir, plots_dir
+            )
+            analysis_stats["run_acceleration_comfort_analysis"] = stats
+        except:
+            analysis_stats["run_acceleration_comfort_analysis"] = None
+
+        # 4. Turn accuracy analysis by spline fitting
+        try:
+            stats, _, _, _, _, _, _ = run_acceleration_comfort_analysis(
+                mcap_path, data_dir, plots_dir
+            )
+            analysis_stats["run_acceleration_comfort_analysis"] = stats
```

Comment:
These don't exist yet, but just showing what it might look like. They will added to next PRs.